client/src/js/services/grid/GridGrouping.js

```diff
@@ -129,31 +138,88 @@ function GridGroupingService(uiGridGroupingConstants, $filter, Session, $timeout
    */
   function configureDefaultGroupingOptions(gridApi) {
 
-    // bind the group selection method
-    gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+    //this instruction block can be executed if the grid involves selection functionality
+    if(gridApi.selection){
+
+      // bind the group selection method
+      gridApi.selection.on.rowSelectionChanged(null, selectAllGroupElements.bind(this));
+
+      gridApi.selection.on.rowSelectionChangedBatch(null, handleBatchSelection.bind(this));
+
+    }
 
     // hook into rows rendered call to ensure the grid is ready before expanding initial nodes
     gridApi.core.on.rowsRendered(null, util.once(function () {
-      gridApi.grouping.groupColumn('trans_id');
+      gridApi.grouping.groupColumn(this.column);
 
       // for the expandAllRows() to be fired last
-      $timeout(gridApi.treeBase.expandAllRows, 0, false);
-    }));
+      unfoldAll(gridApi);
+    }.bind(this)));
+  }
+
+  function unfoldAll(api) {
+    $timeout(api.treeBase.expandAllRows, 0, false);
+  }
+
+  function changeGrouping (column) {
+
+    this.gridApi.grouping.setGrouping(
```

Comment:
I think it would be clearer to use a combination of `clearGrouping()`, `aggregateColumn()` and `groupColumn()` here.  This works (I think, haven't tested), but it feels like a large messy JSON object.  Could you try a more functional style?  [Docs for reference](http://ui-grid.info/docs/#/api/ui.grid.grouping.api:PublicApi).